core/sdk-core/src/test/java/software/amazon/awssdk/core/exception/SdkExceptionTest.java

```diff
@@ -0,0 +1,19 @@
+package software.amazon.awssdk.core.exception;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+
+public class SdkExceptionTest {
+    @Test
+    public void sdkExceptionMessagesDefaultToChildMessages() {
+        Exception barCause = new Exception("bar");
+        Exception nullCause = new Exception();
+
+        assertThat(SdkException.builder().build().getMessage()).isEqualTo(null);
+        assertThat(SdkException.builder().cause(nullCause).build().getMessage()).isEqualTo(null);
+        assertThat(SdkException.builder().message("foo").build().getMessage()).isEqualTo("foo");
+        assertThat(SdkException.builder().message("foo").cause(barCause).build().getMessage()).isEqualTo("foo");
+        assertThat(SdkException.builder().cause(barCause).build().getMessage()).isEqualTo("bar");
```

Comment:
Ideally these would be separate tests so as to follow the 'unit test should test just one behavior' pattern/constraint.

Reply:
Improved